traits/src/registry.rs

```diff
@@ -76,3 +77,81 @@ pub trait AccountIdFor<Assets> {
 	/// Create a name to uniquely identify a share token for given assets and an identifier.
 	fn name(assets: &Assets, identifier: Option<&[u8]>) -> Vec<u8>;
 }
+
+use frame_support::dispatch::Parameter;
+
+pub trait Inspect<AssetNativeLocation, Balance> {
```

Comment:
Why this traits needs generics? it does not seem to use them.

Same for associated type Error.

also AssetId should be probably generic here.

Reply:
I probably added these generics prematurely but I think they will be necessary after we will remove old traits. I will remove it for now.

I wanted to use this one as supretrait for other registry traits that's why I added these associated types.

Why do you think AssetId` should be generic?